javalib/src/main/scala/java/util/concurrent/ConcurrentLinkedQueue.scala

```diff
@@ -0,0 +1,45 @@
+package java.util.concurrent
+
+import java.util._
+
+class ConcurrentLinkedQueue[E]()
+    extends AbstractQueue[E] with Queue[E] with Serializable {
+
+  def this(c: Collection[_ <: E]) = {
+    this()
+    addAll(c)
+  }
+
+  private var inner: LinkedList[E] = new LinkedList()
+
+  override def add(e: E): Boolean =
+    if (e == null) throw new NullPointerException()
+    else inner.add(e)
+
+  override def offer(e: E): Boolean =
+    add(e)
+
+  override def poll(): E =
+    inner.poll()
+
+  override def peek(): E =
+    inner.peek()
+
+  override def isEmpty(): Boolean =
+    inner.isEmpty()
+
+  override def size(): Int =
+    inner.size()
```

Comment:
The contract of `ConcurrentLinkedQueue` says that if the queue has more than `Int.MaxValue` it has to return `Int.MaxValue`. The implementation in `LinkedList` is not currently doing that, but it should be because the `List` interface also defines it as such. 

Look at the case where your linked list has `Int.MAX_VALUE` and you add an element. You will do at at some point a `_size += 1` which will make the size equal to `Int.MIN_VALUE` :(